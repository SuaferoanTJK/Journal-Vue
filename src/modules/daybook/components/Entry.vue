<script>
import { formatDate } from "../helpers/formatDate";

export default {
  name: "Entry-component",
  props: {
    entry: {
      type: Object,
      required: true,
    },
  },
  computed: {
    shortText() {
      return this.entry.text.length > 100
        ? this.entry.text.substring(0, 100) + "..."
        : this.entry.text;
    },
    date() {
      const date = formatDate(this.entry.date);
      return date;
    },
  },
};
</script>

<template>
  <div class="p-4 bg-white rounded my-4 mr-2 shadow-lg border">
    <div
      @click="$router.push({ name: 'entry', params: { id: entry.id } })"
      class="font-semibold text-lg cursor-pointer hover:underline"
    >
      <p>{{ date }}</p>
    </div>
    <p>{{ shortText }}</p>
  </div>
</template>
