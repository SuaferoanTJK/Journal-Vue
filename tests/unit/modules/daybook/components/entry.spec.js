import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Entry Unit Testing", () => {
  const mockRouter = {
    push: jest.fn(),
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Entry, {
      props: {
        entry: journalState.entries[0],
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
  });

  test("Should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Should redirect when it is clicked", () => {
    const entryContainer = wrapper.find(".entryContainer");
    entryContainer.trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: journalState.entries[0].id },
    });
  });
  test("Computed properties test", () => {
    expect(wrapper.vm.date).toBe("Thursday, August 04, 2022");
    expect(wrapper.vm.shortText).toBe("Hi world V2");
  });
});
