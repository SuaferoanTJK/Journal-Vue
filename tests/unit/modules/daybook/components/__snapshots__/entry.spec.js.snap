// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Unit Testing Should match with snapshot 1`] = `
<div class="p-4 bg-white rounded my-4 mr-2 shadow-lg border">
  <div class="entryContainer font-semibold text-lg cursor-pointer hover:underline">
    <p>Thursday, August 04, 2022</p>
  </div>
  <p>Hi world V2</p>
</div>
`;
